trajopt_ifopt/include/trajopt_ifopt/constraints/cartesian_line_constraint.h

```diff
@@ -0,0 +1,193 @@
+﻿/**
+ * @file cartesian_position_constraint.h
+ * @brief The cartesian position constraint
+ *
+ * @author Levi Armstrong
+ * @author Matthew Powelson
+ * @date May 18, 2020
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2020, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TRAJOPT_IFOPT_CARTESIAN_LINE_CONSTRAINT_H
+#define TRAJOPT_IFOPT_CARTESIAN_LINE_CONSTRAINT_H
+
+#include <trajopt_utils/macros.h>
+TRAJOPT_IGNORE_WARNINGS_PUSH
+#include <Eigen/Eigen>
+#include <ifopt/constraint_set.h>
+
+#include <tesseract_kinematics/core/forward_kinematics.h>
+#include <tesseract_environment/core/environment.h>
+#include <tesseract_environment/core/utils.h>
+TRAJOPT_IGNORE_WARNINGS_POP
+
+#include <trajopt/utils.hpp>
+#include <trajopt_ifopt/variable_sets/joint_position_variable.h>
+
+namespace trajopt
+{
+/**
+ * @brief Contains kinematic information for the cartesian position cost; inlude cart point .h & remove?
+ */
+struct CartPosKinematicInfo
```

Comment:
@Levi-Armstrong What do you think here? Having a single struct would be nice. We currently have one for cartesian position and inverse kinematics constraints. We could add an IK and FK object to this and use the same one. I'm not sure how this relates to the command language manipulator info